### ISSUE TITLE
Exclusions start working incorrectly

### ISSUE THREAD

**Issue by unnamed777** [AdguardTeam/AdGuardVPNForAndroid#211]
- **AdGuard VPN version:** 2.2.36
- **VPN mode:** Local VPN
- **Device:** Xiaomi 12S Ultra
- **Operating system and version:** Android 12
- **Root access:** No

I use exclusions (selected mode) and apps settings. In exclusion list I have instagram.com, facebook.com and all their domains (the services are blocked by my provider). In apps settings I have app "Instagram" enabled. Yesterday I've installed update of AdGuard VPN from Play Store and Instagram started working incorrectly on my phone and tablet. When VPN is enabled and exclusions is in selected mode, instagram feed is loaded, but media is not (no photos, no videos, no avatars, no images in stories). I can open user profile, but media isn't loaded. When I set exclusions to general mode with empty list (in other words, everything goes through VPN), Instagram's media is loaded and the app works perfectly as before the adguard update.

Either the last update brought an issue or Instagram recently introduced new cdn domain, which should be added to the exclusion list. I've sent app logs ("extreme" log level) to devteam@adguard.com. If there is no problem in the logs, could you please suggest a way to determine which domains have to be added to the exclusions? It would be great to have some diagnostic interface in adguard app for such tasks (list of last bypassed domains, for instance).

**Comment by ameshkov** [comment]
Do you have just `instagram.com` on the list or "Instagram" as a service?

**Comment by unnamed777** [comment]
As a service, Facebook too.

**Comment by ameshkov** [comment]
Yeah, there's a chance it's using more domains now, we'll need to check that.

**Comment by unnamed777** [comment]
Yes.

**Comment by maxikuzmin** [comment]
@unnamed777 do you have AdGuard on your device?

**Comment by unnamed777** [comment]
No.

**Comment by unnamed777** [comment]
Any ideas?

**Comment by maxikuzmin** [comment]
@unnamed777 we are looking into this issue, we need a little more time. Please wait.

**Comment by maxikuzmin** [comment]
Does @unnamed777 only VPN work or are you using AdGuard + AdGuard VPN in integration mode?

**Comment by unnamed777** [comment]
I'm using AdGuard VPN only.

> Also try in General mode without exclusions for Instagram and Facebook to check. Will Instagram stay the same behavior with this?
I'm already using this mode (general mode, no exclusions at all, Instagram is enabled in "Apps settings"), as I described in the issue. Instagram works in this mode.

**Comment by artemiv4nov** [comment]
@unnamed777 Hello!
Please add the following domains to exclusions:
```
cdn.fbsbx.com
cdninstagram.com
facebook.com
facebook.net
fbcdn.net
instagram.com
```

Maybe it helps.

**Comment by unnamed777** [comment]
All of these domains are already included in "Instagram" and "Facebook" services. These services are enabled in Selected mode.
<details>
<summary>Screenshots</summary>
![Screenshot_2022-11-24-08-01-48-894_com adguard vpn](https://user-images.githubusercontent.com/1111833/203675691-69ff22c2-7ebc-48d8-8907-73798212ee86.jpg)
![Screenshot_2022-11-24-08-01-55-548_com adguard vpn](https://user-images.githubusercontent.com/1111833/203675696-9dcfa8de-e45f-4357-a994-06f500fa8269.jpg)
![Screenshot_2022-11-24-08-02-00-026_com adguard vpn](https://user-images.githubusercontent.com/1111833/203675705-adf73c9c-18cb-4cde-82de-75c49a5d8559.jpg)
</details>

**Comment by unnamed777** [comment]
All of these domains are already included in "Instagram" and "Facebook" services. These services are enabled in Selected mode.

![Screenshot_2022-11-24-08-01-48-894_com adguard vpn](https://user-images.githubusercontent.com/1111833/203676042-cbf0005c-b034-41ed-acf8-bef59f8918dc.jpg)
![Screenshot_2022-11-24-08-01-55-548_com adguard vpn](https://user-images.githubusercontent.com/1111833/203676043-736016c8-8b78-4124-9203-c14fc36d4c29.jpg)
![Screenshot_2022-11-24-08-02-00-026_com adguard vpn](https://user-images.githubusercontent.com/1111833/203676048-fe3c9ae0-126e-466a-8505-3da03e4b1b19.jpg)

**Comment by maxikuzmin** [comment]
@unnamed777 to troubleshoot this issue, we need to get additional logs.

Here's what we need you to do:

- Collect the debug log as it's [explained here](https://kb.adguard.com/en/android/solving-problems/log#debug);
- Remember the exact time when the issue was reproduced. We will need it to find the corresponding records in the log file;
- Additionally, capture the system bug report after reproducing the issue. [Here's the instruction](https://developer.android.com/studio/debug/bug-report);
- Send both files to devteam@adguard.com. Mention the Github issue number and the exact time when the issue was reproduced.

**Comment by unnamed777** [comment]
I already sent such log when I created the issue. Have you received email with subject "Logs for AdGuardVPNForAndroid #211"? Do you need a new log?

**Comment by maxikuzmin** [comment]
@unnamed777 use "General mode" for now, we are working on your problem

**Comment by ameshkov** [comment]
@maxikuzmin should we move this to VpnLibs?
